yapapi/contrib/strategy/provider_filter.py

```diff
@@ -71,9 +71,10 @@ def __init__(self, base_strategy: BaseMarketStrategy, is_allowed: IsAllowedType)
         self._is_allowed = is_allowed
 
     async def score_offer(self, offer: OfferProposal) -> float:
-        allowed = self._is_allowed(offer.issuer)
-        if inspect.iscoroutinefunction(allowed):
-            allowed = await allowed  # type: ignore
+        if inspect.iscoroutinefunction(self._is_allowed):
+            allowed = await self._is_allowed(offer.issuer)  # type: ignore
+        else:
+            allowed = self._is_allowed(offer.issuer)
```

Comment:
```suggestion
            allowed = self._is_allowed(offer.issuer)
            if inspect.isawaitable(allowed):
                allowed = await allowed
```

would save us from having to keep two calls to `self._is_allowed`